requirements.txt

```diff
@@ -46,3 +46,4 @@ tenacity>=8.3.0,!=8.4.0
 # the below library should NOT be imported into any python files
 # it is for CLI usage ONLY
 yamllint>=1.35.1
+referencing>=0.35.0,<0.36.0
```

Comment:
This should be brought in by `jsonschema`. Not sure why it should be explicit here.

Reply:
I assume that if it's directly imported, then it's a direct dependency.

Reply:
Don't pin upper version unless a higher version introduces a known breaking change.